Rymapt/ethoscope
# Standardise tests

A few fixes to the tests

* Switch to unittest for consistency with the tests in `src`
* Changed the directory structure in `node_src` to match the structure in `src` (`test` -> `tests/unittests`).
* Use a mock SQL database for the MySQLdbCSVWriter tests
* Make sure paths are absolute so that it doesn't matter what directory tests are started in

The tests can be run by executing
```sh
python -m unittest discover
```
from any directory in the source tree.  Note that `/node_src` doesn't have a `__init__.py` file so the node tests can only be discovered if running in `node_src` or a subdirectory.  The device `/src` directory has an init file, so those can be discovered from the base directory.  This can be changed if required?
